src/components/PdfButton/PdfButton.cy.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import PdfButton from "@/components/PdfButton/PdfButton";
+import ShippingLabelsPdf, { ParcelClients } from "@/pdfs/ShippingLabels/ShippingLabelsPdf";
+
+const downloadsFolder = Cypress.config("downloadsFolder");
+const fileName = "ShippingLabels.pdf";
+const parcelClientsData: ParcelClients[] = [
+    {
+        packing_datetime: "19/06/2023",
+        collection_centre: "Delivery",
+        collection_datetime: "31/07/2023, 14:00",
+        voucher_number: "VOUCHER-NUMBER-2",
+        full_name: "Harry Potter",
+        phone_number: "07987654321",
+        address_1: "Hogwarts Castle",
+        address_2: "",
+        address_town: "Scottish Highlands",
+        address_county: "United Kingdom",
+        address_postcode: "YE11OW",
+        delivery_instructions: "Owls only no floo powder please.",
+        index: 1,
+        total: 2,
+    },
+    {
+        packing_datetime: "20/06/2023",
+        collection_centre: "Delivery",
+        collection_datetime: "31/08/2023, 14:00",
+        voucher_number: "VOUCHER-NUMBER-2",
+        full_name: "Nessie",
+        phone_number: "0799999999",
+        address_1: "Loch Ness",
+        address_2: "Inverness",
+        address_town: "Scottish Highlands",
+        address_county: "United Kingdom",
+        address_postcode: "NE55IE",
+        delivery_instructions: "Throw it in the water and leave as quickly as possible.",
+        index: 2,
+        total: 2,
+    },
+];
+
+const allFields = parcelClientsData
+    .map((obj) => {
+        return Object.values(obj);
+    })
+    .flat();
+
+describe("Export Pdf Button", () => {
+    it("Renders", () => {
+        cy.mount(
+            <PdfButton
+                text="Click"
+                fileName="ShippingLabels.pdf"
+                data={parcelClientsData}
+                Children={ShippingLabelsPdf}
+            />
+        );
+    });
+
+    it("File is saved", () => {
+        cy.mount(
+            <PdfButton
+                text="Click"
+                fileName="ShippingLabels.pdf"
+                data={parcelClientsData}
+                Children={ShippingLabelsPdf}
+            />
+        );
+
+        cy.get("a").click();
+        cy.readFile(`${downloadsFolder}/${fileName}`);
+    });
+
+    it("All data of shipping label is included in the file", () => {
+        cy.mount(
+            <PdfButton
+                text="Click"
+                fileName="ShippingLabels.pdf"
+                data={parcelClientsData}
+                Children={ShippingLabelsPdf}
```

Comment:
Not sure how feasible this is, but a test to check the page breaks might be good?

Reply:
Might be overkill tho

Reply:
Yes I agree that its overkill.